frontend/src/metabase/dashboard/components/Dashboard/Dashboard.tsx

```diff
@@ -258,7 +258,10 @@ function Dashboard(props: DashboardProps) {
       });
 
       if (!isSuccessfulFetchDashboardResult(result)) {
-        setErrorPage(result.payload);
+        const payload = result.payload as { isCancelled?: boolean };
```

Comment:
Please use a type guard instead of a cast - `result.payload` is `unknown` and this code can fail in case `payload` is `null` or `undefined`.